src/container.ts

```diff
@@ -486,7 +493,14 @@ export function createContainer<T extends object = any, U extends object = any>(
           return undefined
         }
 
-        throw new AwilixResolutionError(name, resolutionStack)
+        const defaultArgs = getDefaultArgs()
```

Comment:
Wouldn't this just be `const defaultArgs = resolver.defaultArgs`?

Reply:
No because `resolver` in this case refers to the current thing being resolved (`unresolved` in the test cases). As a result we need to get the parent of the current item being resolved and get the `defaultArgs` from that one.